### PR TITLE
Cache ansible inventory data

### PR DESCRIPTION
**Pull request by perlun**
Invoking `ansible` and `ansible-inventory` on each completion incurs a delay of at least 300 + 300 ms. In my experience, the actual latency is even worse (more like 700-800 ms). Now, this might seem a tolerable latency for some, but for something that you execute interactively, while typing a command, it's far too much for my taste.

The naïve approach to fix this here is to just cache this output in a `.txt` file and `cat` this file on subsequent invocations. It improves the performance significantly, going down to ~3 ms on my local machine.

Downside: cache invalidation. Whenever a host or group is added/removed to/from the Ansible inventory, the cached content would have to be invalidated. A simple approach would be to just delete it in a `cron` job, but in my experience it's very likely that you want to be able to execute one or more Ansible commands pretty much immediately after
updating the inventory. I think I will probably just make an `ansible_clear_cache` alias locally to delete the file and execute it
when I need it. :smile: 

### Future improvements

A more advanced approach would be to check the timestamp on the file and only allow it to be used until a given TTL has expired. (Yes, we are going into overingineering now.) Even better: kick of a background process to update the cached version anytime completion has been performed. This would require some thinking to get it done right, though.

### CODE REVIEW

File: ansible-completion.bash [hunk 1]
```diff
@@ -30,10 +30,17 @@ _ansible_complete_host() {
...
-    local hosts_and_groups=$(ansible all --list-hosts 2> /dev/null | tail +2 &&
-        ansible-inventory --list | jq -r "keys | .[]")
+    if [[ ! -f "$HOME/.cache/ansible-completion/ansible_inventory_content.txt" ]]; then
+        mkdir -p $HOME/.cache/ansible-completion
+
+        # The 'tail' call is there to get rid of the first 'hosts (xx)' line in
+        # the output. 'awk' call is there to trim leading and trailing
+        # whitespace from the output.
+        ansible all --list-hosts 2> /dev/null | tail +2 | awk '{$1=$1};1' > $HOME/.cache/ansible-completion/ansible_inventory_content.txt
+        ansible-inventory --list | jq -r "keys | .[]" >> $HOME/.cache/ansible-completion/ansible_inventory_content.txt
```
Comment (slovdahl) [inline ansible-completion.bash]: How does this work with multiple Ansible repositories on the same host? :thinking:

Comment (perlun) [reply ansible-completion.bash]: Poorly, likely. :unamused: It doesn't really scale beyond a single `ansible.cfg` AFAICT. I wonder how we could make it be more generally usable in that sense. One way would be to be able to override the `$HOME/.cache/ansible-completion`-path using an ENV variable or something. Any other ideas?

Comment (slovdahl) [reply ansible-completion.bash]: Add `$CWD` to the name of the file in `.cache`?

Comment (perlun) [reply ansible-completion.bash]: Hmm, `$CWD` doesn't seem to be defined in my `bash` at least. Are you referring to the full `pwd` or something like `basename $(pwd)`?

Comment (slovdahl) [reply ansible-completion.bash]: Yeah, the full `$(pwd)` or something like that. Slap a `md5sum` around it to make it safe for file name usage?

Comment (perlun) [reply ansible-completion.bash]: Thanks. I added an implementation now (had to pipe it through `awk` to get rid of some extra noise that `md5um` prints out). With these changes in place, it creates the file like this:

```
$ tree ~/.cache/ansible-completion/
/home/per/.cache/ansible-completion/
└── 50b28211a61d16b65ceb2977e2f8a2a9
    └── ansible_inventory_content.txt

1 directory, 1 file
```

Comment (perlun) [reply ansible-completion.bash]: @slovdahl Anything holding this back from approval now?

Comment (slovdahl) [reply ansible-completion.bash]: No, I didn't realize you wanted my approval :joy:

**Review by slovdahl — APPROVED** [review]

**Pull request merged**